add test if  roxygen has been run

waddella opened the pull request:
Hey @gmbecker,

I have added a simple circleci check to see whether the manuals, DESCRIPTION and NAMESPACE have been update.

waddella commented:
OK, now the checks fail if there is an error. currently I don't fail the pipeline if there are warnings or notes.

gmbecker commented on .circleci/config.yml:
```diff
@@ -26,16 +26,38 @@ jobs:
       - store_test_results:
           path: rtables.Rcheck/tests
 
+  roxygen:
```
@waddella this doesn't appear to work, going into circleci I see:

```
> roxygen2::roxygenize('.', roclets = c('rd', 'collate', 'namespace'))
Warning: Version of roxygen2 last used with this package is 7.1.0.  You only have version 7.0.2
Writing NAMESPACE
Loading rtables
Loading required package: magrittr
Creating a generic function for ‘nrow’ from package ‘base’ in package ‘rtables’
Creating a generic function for ‘ncol’ from package ‘base’ in package ‘rtables’
Creating a generic function for ‘rbind’ from package ‘base’ in package ‘rtables’
Writing DM.Rd
Writing NAMESPACE
> 
> 
manuals are upto date with roxygen comments
```

The "Writing DM.Rd" suggests that the rd files were not up to date and thus this checek should have failed, correct?

gmbecker commented on man/add_analyzed_colvars.Rd:
```diff
@@ -34,7 +34,7 @@ library(dplyr)
 
 # TODO: fix
-build_table(l, ANL)
+# build_table(l, ANL)
```
I believe this is fixed, please (re)base your pr on the current state of master

waddella replied:
I did that, thanks

gmbecker commented on man/add_colby_multivar.Rd:
```diff
@@ -34,7 +34,7 @@ library(dplyr)
 
 # TODO: fix
- build_table(l, ANL)
+# build_table(l, ANL)
```
As above, I believe this is fixed so please rebase (or whatever the term is) the PR.

gmbecker commented on R/colby_constructors.R:
```diff
@@ -371,7 +371,7 @@ add_rowby_varlevels = function(lyt,  var, lbl = var,  vlblvar = var, splfun = NU
 #' 
 #' # TODO: fix
-#'  build_table(l, ANL)
+#' # build_table(l, ANL)
```
These should work now.

gmbecker commented on .circleci/config.yml:
```diff
@@ -26,16 +26,38 @@ jobs:
-#      - store_artifacts:
-#          path: rtables.Rcheck
-#          destination: rtables.Rcheck
+      - store_artifacts:
```
I think we want the install output and if possible even the build output as well as just 00check.log

waddella replied:
That information is available in the Steps:

![image](https://user-images.githubusercontent.com/7786462/80275183-d524a900-86df-11ea-9aa9-79c32300c063.png)

I will remodel the the pipeline to group them nicer together and I will also add the install output from the .Rcheck folder.

waddella commented:
I am currently working on this, I will let you know once I am done

waddella commented:
@gmbecker, please see my changes. I have updated the automation pipelines, added the `redesign/aw_play.R` file (notes from the meeting. I have also fixed `Viewer()` and have added some other minor changes. Please squash merge, the commit log is not pretty :smile:

gmbecker commented on .circleci/config.yml:
```diff
@@ -1,73 +1,97 @@
+
 
-  check:
+  roxygen:
```
rd-up-to-date checking should come before R CMD check and end the pipeline if it fails because doing the expensive check is pointless if the checks  aand example running isn't happening on the correct files anyway

gmbecker commented:
@waddella  I spun out ta couple  of things as individual issues, but overall looks good so I've merged it.

gmbecker commented:
I  spun out a couple o f things as individual issues but overall looks good so I've merged it.
